scipy/spatial/ckdtree/src/remove.cxx

```diff
@@ -0,0 +1,71 @@
+#include <vector>
+#include "ckdtree_decl.h"
+#include <math.h>
+
+int
+find(ckdtree *self, ckdtreenode *actual_node, const double *point, std::vector<ckdtreenode *> *node_path){
+   while(actual_node->split_dim != -1){
+        if(*(point + actual_node->split_dim) < actual_node->split)
+            actual_node = actual_node->less;
+        else
+            actual_node = actual_node->greater;
+
+        node_path->push_back(actual_node);
+    }
+    //now actual_node is the node that can contains point
+
+    for(int idx_indices=actual_node->start_idx; idx_indices<actual_node->end_idx; idx_indices++){
+        bool equal = true;
+        int idx_data = *(self->raw_indices + idx_indices);
+
+        for(int point_dimension=0; point_dimension<self->m; point_dimension++){
+            double *value_of_p = self->raw_data + idx_data*self->m + point_dimension;
+            if(*value_of_p != *(point+point_dimension)){
```

Comment:
Requiring exact floating point equality seems error prone and could be why the tests are failing. Would it not make more sense to delete by index? What is the use case you had in mind here?

Reply:
> Requiring exact floating point equality seems error prone and could be why the tests are failing. Would it not make more sense to delete by index? What is the use case you had in mind here?

Uhm, good observation! I was thinking about this: if someone has only the point, before removing by index, he has to find index. To find the index the best way is querying 1-nn and then check the equality. On the other hand, if someone has the index, can simply remove self.data[index]. 

It is a little difference, so if you think that removing by index is better i can adapt the code easly!

Reply:
Removing by index seems more robust to me. e.g. It's unclear what this should do when there are multiple points nearby. Right now it just picks a random point within eps of the query point which probably isn't what you want.